husky-misc/code-challenge
# Code Challenge: Store and Compute

### Overview

This PR contains the source code of this code challenge and bellow I'm going to explain how to run and test that.

### Step by Step

This project was done with MongoDB and Rails API, so it's required to have those software in your machine:

- Ruby 2.3.4
- MongoDB

To start testing that locally you need to follow those steps:
```
> clone this repository
> make sure if MongoDB is running in default port (if need to change -> config/mongoid.yml)
> bundle install
> rake seeds
> rails s
```

### How to run tests

```
> make sure MongoDB is running locally in default port (if need to change -> config/mongoid.yml)
> rspec
```

### Endpoints

| Endpoint                                                              | Description
-----------------------------------------------------------------|------------------
POST /api/v1/store                                                | Store ips with values
POST /api/v1/compute                                          | Compute non-computed ips
GET /api/v1/compute_histories?per=&page=       | List computation history

### How to authenticate

Just set those headers
```
Content-Type: application/json; charset=utf-8
Authorization: Bearer ed62fc4bc51f2ab94b59c15f6c9f63896dce60dc
```

### Resources

API URL: https://protected-lowlands-51884.herokuapp.com/
Client URL: https://husky-c366e.firebaseapp.com/
Client GH Project: https://github.com/paulinhamp/code-challenge-client
